docs/src/architecture/notes.md

```diff
@@ -1,73 +1,191 @@
-
-In `miden-lib` there are predefined P2ID and P2IDR note scripts that every user can simply create, see [code](https://github.com/0xPolygonMiden/miden-base/blob/fa63b26d845f910d12bd5744f34a6e55c08d5cde/miden-lib/src/notes/mod.rs#L15-L66). The script ensures that only the specified target account can consume the note and receive the asset. 
-
-The Note scripts is also the root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) which means every function is a commitment to the underlying code. The code cannot change unnoticed to the user because its hash would change.
-
-# Note recipient 
-We define `recipient` as: `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)` represented as `Word`. The account that executes a given note needs to provide the pre-image data to `recipient` during the transaction prologue. That means, one can create notes that can only be consumed if the `serial_num` and other data is known. This information can be passed on off-chain by the sender to the recipient. 
-
-# Note storage modes
-Similar to accounts, there are two storage modes for notes in Miden. Notes can be stored privately in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) with only the note hash. Or notes can be stored publicly with all data.
-
-Privately stored notes can only be consumed if the note data is known to the consumer. That means, there must be some offchain communication to transmit the note's data from the sender to the receipient.
-
-# Note metadata
-For every note the Miden Operator stores metadata in the Note DB. This metadata includes:
-
-* A **user-defined tag** as a means to quickly grab all notes for a certain application or use case.
-* A **sender** to be able to provide also ERC20 contract functionality.
-* The **number of assets** contained in the note. 
-
-# Note hash
-The note hash is computed as:
+ In addition, a note has **metadata** including the sender and the note tag.
 
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+# Note's lifecycle
+New notes are created by executing transactions. After verifying the transaction proof the operator adds either only the note hash (private notes) or the full note data (public notes) to the [Note DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database). Notes can be produced and consumed locally by users in local transactions or by the operator in a network transaction. Note consumption requires the transacting party to know the note data to compute the nullifier. After successful verification, the operator sets the corresponding entry in the Nullifier DB to `1` (=consumed). 
```

Comment:
I changed the part with the nullified. 

I suggest, as long as we don't have encrypted on-chain data, we keep the terms private and public for off-chain and on-chain as it is easier to understand for users and still correct. 

Would you agree?

Reply:
I would switch to this terminology now. I think it will be much more work to establish one terminology now and then switch to a different terminology in several months. Also, we'll probably need to use the terminology for encrypted notes much sooner than we actually implement them.